Podcast/DiscoverViewController.swift

```diff
@@ -2,202 +2,283 @@
+        let episodeDetailViewController = EpisodeDetailViewController()
+        episodeDetailViewController.episode = topEpisodes[indexPath.row]
+        navigationController?.pushViewController(episodeDetailViewController, animated: true)
     }
 
-    func recommendedEpisodeOuterTableViewCellDidPressPlayButton(episodeTableViewCell: EpisodeTableViewCell, episode: Episode) {
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+}
+
+// MARK: - Episode Table View Cells
+extension DiscoverViewController: EpisodeTableViewCellDelegate {
+    func episodeTableViewCellDidPressPlayPauseButton(episodeTableViewCell: EpisodeTableViewCell) {
+        guard let episodeIndexPath = topEpisodesTableView.indexPath(for: episodeTableViewCell), let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+        let episode = topEpisodes[episodeIndexPath.row]
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: episode)
+        episodeTableViewCell.updateWithPlayButtonPress(episode: episode)
+
+        // reset previously playings view
+        if let playingIndexPath = currentlyPlayingIndexPath, currentlyPlayingIndexPath != episodeIndexPath, let currentlyPlayingCell = topEpisodesTableView.cellForRow(at: playingIndexPath) as? EpisodeTableViewCell {
+            let playingEpisode = topEpisodes[playingIndexPath.row]
+            currentlyPlayingCell.updateWithPlayButtonPress(episode: playingEpisode)
+        }
+
+        // update index path
+        currentlyPlayingIndexPath = episodeIndexPath
     }
-    
-    func recommendedEpisodeOuterTableViewCellDidPressBookmarkButton(episodeTableViewCell: EpisodeTableViewCell, episode: Episode) {
+
+    func episodeTableViewCellDidPressRecommendButton(episodeTableViewCell: EpisodeTableViewCell) {
+        guard let episodeIndexPath = topEpisodesTableView.indexPath(for: episodeTableViewCell) else { return }
+        let episode = topEpisodes[episodeIndexPath.row]
+        episode.recommendedChange(completion: episodeTableViewCell.setRecommendedButtonToState)
+    }
+
+    func episodeTableViewCellDidPressBookmarkButton(episodeTableViewCell: EpisodeTableViewCell) {
+        guard let episodeIndexPath = topEpisodesTableView.indexPath(for: episodeTableViewCell) else { return }
+        let episode = topEpisodes[episodeIndexPath.row]
         episode.bookmarkChange(completion: episodeTableViewCell.setBookmarkButtonToState)
     }
-    
-    func recommendedEpisodeOuterTableViewCellDidPressRecommendButton(episodeTableViewCell: EpisodeTableViewCell, episode: Episode) {
-        episode.recommendedChange(completion: episodeTableViewCell.setRecommendedButtonToState)
+
+    func episodeTableViewCellDidPressTopicButton(episodeTableViewCell: EpisodeTableViewCell, index: Int) {
+        //        guard let episodeIndexPath = topEpisodesTableView.indexPath(for: episodeTableViewCell) else { return }
```

Comment:
If I'm not mistaken you can delete this required function from EpisodeTableViewCellDelegate! We don't have topics for episodes right now.